lib/devmapper_mock.ml

```diff
@@ -324,68 +197,101 @@ module DeviceSet = struct
-  try
-    ref (DeviceSet.load_file !persistent_fn)
-  with _ ->
-    ref (DeviceSet.make ())
+  let create device targets =
+    DeviceSet.create !devices device targets;
+    DeviceSet.save_file !devices !persistent_fn
 
-let create device targets =
-  DeviceSet.create !devices device targets;
-  DeviceSet.save_file !devices !persistent_fn
+  let remove device =
+    DeviceSet.remove !devices device;
+    DeviceSet.save_file !devices !persistent_fn
 
-let remove device =
-  DeviceSet.remove !devices device;
-  DeviceSet.save_file !devices !persistent_fn
+  let reload device targets =
+    DeviceSet.reload !devices device targets;
+    DeviceSet.save_file !devices !persistent_fn
 
-let reload device targets =
-  DeviceSet.reload !devices device targets;
-  DeviceSet.save_file !devices !persistent_fn
+  let suspend device =
+    DeviceSet.suspend !devices device;
+    DeviceSet.save_file !devices !persistent_fn
 
-let suspend device =
-  DeviceSet.suspend !devices device;
-  DeviceSet.save_file !devices !persistent_fn
+  let resume device =
+    DeviceSet.resume !devices device;
+    DeviceSet.save_file !devices !persistent_fn
 
-let resume device =
-  DeviceSet.resume !devices device;
-  DeviceSet.save_file !devices !persistent_fn
+  let mknod device path mode =
+    DeviceSet.mknod !devices device path mode;
+    DeviceSet.save_file !devices !persistent_fn
 
-let mknod device path mode =
-  DeviceSet.mknod !devices device path mode;
-  DeviceSet.save_file !devices !persistent_fn
+  let stat device =
+    let ret = DeviceSet.stat !devices device in
+    DeviceSet.save_file !devices !persistent_fn;
+    ret
 
-let stat device =
-  let ret = DeviceSet.stat !devices device in
-  DeviceSet.save_file !devices !persistent_fn;
-  ret
+  let ls () =
+    let ret = DeviceSet.ls !devices () in
+    DeviceSet.save_file !devices !persistent_fn;
+    ret
 
-let ls () =
-  let ret = DeviceSet.ls !devices () in
-  DeviceSet.save_file !devices !persistent_fn;
-  ret
+  let clear () =
+    DeviceSet.clear !devices ();
+    DeviceSet.save_file !devices !persistent_fn
 
-let clear () =
-  DeviceSet.clear !devices ();
-  DeviceSet.save_file !devices !persistent_fn
+  let get_persistent () =
+    !persistent_fn
 
-let get_persistent () =
-  !persistent_fn
+  let set_persistent filename =
+    persistent_fn := filename
 
-let set_persistent filename =
-  persistent_fn := filename
+  let save_file path =
+    DeviceSet.save_file !devices path
 
-let save_file path =
-  DeviceSet.save_file !devices path
+  let load_file path =
+    devices := DeviceSet.load_file path
+end
+
+let create = Mock.create
+let remove = Mock.remove
+let reload = Mock.reload
+let suspend = Mock.suspend
+let resume = Mock.resume
+let mknod = Mock.mknod
+let stat = Mock.stat
+let ls = Mock.ls
+let clear = Mock.clear
+let get_persistent = Mock.get_persistent
+let set_persistent = Mock.set_persistent
+let save_file = Mock.save_file
+let load_file = Mock.load_file
+
+module type DM_intf = sig
```

Comment:
I think we should adopt a common OCaml convention and:

1. Modify the _oasis to set 'Pack: true'. For example: https://github.com/mirage/mirage-block-volume/blob/14858d04202744c2f9758f1bf0170869562705ab/_oasis#L22
2. Create a file called "s.mli" to contain the signatures. Move this signature to s.mli and rename it to "DEVMAPPER", a bit like this: https://github.com/mirage/mirage-block-volume/blob/14858d04202744c2f9758f1bf0170869562705ab/lib/s.mli#L120
3. Rename the module 'Devmapper_mock' to 'Mock'
4. Rename the module 'Devmapper' to 'Linux'
5. in the .mli files for mock.ml and linux.ml, you can include S.DEVMAPPER to inherit all the bindings e.g. https://github.com/mirage/mirage-block-volume/blob/14858d04202744c2f9758f1bf0170869562705ab/lib/vg.mli#L61

Clients will be able to
```
open Devmapper.Linux
```
or
```
open Devmapper.Mock
````
In future we could have other implementations (perhaps `Devmapper.Mirage`?)